ansible/roles/test/tasks/snmp/lldp.yml

```diff
@@ -31,10 +31,11 @@
     - lldpLocSysName
     - lldpLocSysDesc
 
+# Gather physical interfaces from SNMP
 - set_fact:
     snmp_ports:  "{{ snmp_ports|default({}) | combine({ item.key : item.value}) }}"
   when: "{{ item.value.name is defined }}
-    and {{ item.value['name'].find('Ethernet') != -1 }}"
```

Comment:
>item.value['name'].find('Ethernet') [](start = 11, length = 35)

Actually the original expression is better. We will not depend on ifindex range.